src/tasks/AppleAppBuilder/Templates/runtime.m

```diff
@@ -201,6 +204,50 @@
 //%DllMap%
 }
 
+int32_t GlobalizationNative_LoadICUData(void * pData);
+
+static int32_t load_icu_data()
+{
+    char path [1024];
+    int res;
+
+    const char *dname = "icudt.dat";
+    const char *bundle = get_bundle_path ();
+
+    os_log_info (OS_LOG_DEFAULT, "Loading ICU data file '%s'.", dname);
+    res = snprintf (path, sizeof (path) - 1, "%s/%s", bundle, dname);
+    assert (res > 0);
+
+    // TO DO: Handle the case when the file isn't there.
+    FILE *fp = fopen(path, "rb");
+    if (fp != NULL) {
+        if (fseek(fp, 0L, SEEK_END) == 0) {
+            long bufsize = ftell(fp);
```

Comment:
We should just call `fstat` instead, assuming it's available. It's simpler, more reliable, and comes with fewer caveats.

Reply:
Agreed, but fstat was continually giving me -1 return values claiming the file descriptor was bad (even though it wasn't).

Reply:
Hmm, curious. Let's leave it this way for now and open an issue to try and switch over then, assuming you're sure the ICU data file will never be >4GB (in which case this will break).